.circleci/config.yml

```diff
@@ -37,9 +44,32 @@ jobs:
           name: make
           command: cmake --build $IROHA_BUILD -- -j4
       - run:
+          name: initialize coverage reporting
+          command: lcov --capture --initial --directory $IROHA_BUILD --output-file $IROHA_COVERAGE
+      - run:
           name: run tests
           command: cmake --build $IROHA_BUILD --target test
       - run:
+          name: reporting coverage
+          command : |
+            lcov --capture --directory $IROHA_BUILD --output-file $IROHA_COVERAGE
+            IROHA_COVERAGE_TARGET_DIRECTORIES=('irohad' 'iroha-cli' 'libs')
+            for directory in ${IROHA_COVERAGE_TARGET_DIRECTORIES[@]}; do
+              lcov --extract $IROHA_COVERAGE "$IROHA_HOME/$directory/*" > ${IROHA_COVERAGE}_$directory
+            done
+
+            BASE_COVERAGE_REPORT=${IROHA_COVERAGE_TARGET_DIRECTORIES[0]}
+            IROHA_COVERAGE_TARGET_DIRECTORIES=(${IROHA_COVERAGE_TARGET_DIRECTORIES[@]:1})
+            for directory in ${IROHA_COVERAGE_TARGET_DIRECTORIES[@]}; do
+              lcov --add-tracefile ${IROHA_COVERAGE}_$directory --add-tracefile ${IROHA_COVERAGE}_$BASE_COVERAGE_REPORT > $IROHA_COVERAGE
```

Comment:
It seems that **IROHA_COVERAGE** is same for all cycles of the loop. Is it the right behavior, it seems they're overwriting each other?